admin/environment-management/ssh-access.md

```diff
@@ -1,17 +1,113 @@
 ---
-title: "Disable SSH Access"
-description: Learn how to disable SSH to comply with ingress restrictions.
+title: "SSH Configuration"
+description: Learn how to configure SSH access to Coder environments.
 ---
 
-By default, Coder enables SSH access for all users. There are no additional
-dependencies required.
+By default, Coder enables SSH access for all users. Coder assigns each user a
+private key that they can use to access their environments.
+
+## Background
+
+Part of the standard Coder environment asset bundle is a lightweight SSH server
+mounted onto the environment agent; the lightweight SSH server is a backup used
+when Coder can't find a server available on port 22. This allows slimmer images
+to remain accessible via SSH without the need for additional image dependencies.
+
+## Using OpenSSH
+
+The built-in SSH server is limited and does not correctly implement
+advanced functionality like X11 forwarding or `sshd_config` specifications.
+If SSH is the primary mode of access to Coder for your users, consider
+running a full OpenSSH server with `systemd` inside your image instead.
+
+To do so:
+
+1. Make sure that you're creating your environments with the [CVM
+   option](https://coder.com/docs/environments/cvms) enabled
+2. Add the following to your Dockerfile:
+
+```Dockerfile
+FROM ubuntu:20.04
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    systemd \
+    openssh-server
+
+# Start OpenSSH with systemd
+RUN systemctl enable ssh
+
+# recommended: remove the system-wide environment override
+RUN rm /etc/environment
+
+# recommended: adjust OpenSSH config
+RUN echo "PermitUserEnvironment yes" >> /etc/ssh/sshd_config && \
+  echo "X11Forwarding yes" >> /etc/ssh/sshd_config && \
+  echo "X11UseLocalhost no" >> /etc/ssh/sshd_config
+
+```
+
+> If Coder detects a running TCP server on port 22, it will forward incoming SSH
+> traffic to this server.
+
+At startup, Coder injects the user's SSH key into `~/authorized_keys` to
+facilitate authentication with OpenSSH. For the best experience, add the
+following to your `/etc/ssh/sshd_config` file:
```

Comment:
To my local `.ssh` or to the `ssh` directory in my Coder environment? May want to clarify, but I'm JNP'ing 😄